RavenDB-25594 - Expose SchemaDefinitions property of indexes in Studio

M4xymm opened the pull request:
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-25594

### Additional description

I added validation for a unique collection name to TODO, and then I will add validation for correct JSON to aceEditor.

<img width="2279" height="1089" alt="image" src="https://github.com/user-attachments/assets/9dd409c8-cb95-48e6-ae49-9de39557a852" />

<img width="1491" height="275" alt="image" src="https://github.com/user-attachments/assets/725f5acd-d414-4715-b0e1-4d6029f23c46" />


### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [x] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [x] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing
- [ ] Existing tests verify the correct behavior

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

M4xymm committed:
RavenDB-25594 feat (schema definitions): Introduced support for schema definitions in index creation and editing

- Added `schemaDefinitionModel` for managing schema definitions.
- Updated index editor to support adding, formatting, and deleting schema definitions.
- Replaced the single `schemaDefinitionsText` field with an observable array of schemaDefinitionModel instances for better structure and management.

RavenDB-25594 feat (schema definitions): Enhanced schema editor with prettify JSON function and refined collection handling. Improved UI for schema definitions.

kalczur commented on src/Raven.Studio/typescript/common/generalUtils.ts:
```diff
@@ -814,6 +814,14 @@ class genUtils {
         return Number(tokens).toLocaleString();
     }
+
+    static prettifyContent(content: string, spacing: number = 2) {
```
```suggestion
    static prettifyJson(content: string, spacing: number = 2) {
```

kalczur commented on src/Raven.Studio/typescript/common/generalUtils.ts:
```diff
@@ -814,6 +814,14 @@ class genUtils {
+    static prettifyContent(content: string, spacing: number = 2) {
+        try {
+            return JSON.stringify(JSON.parse(content), null, spacing);
+        } catch (e) {
```
```suggestion
        } catch {
```

kalczur commented on src/Raven.Studio/typescript/models/database/index/indexDefinition.ts:
```diff
@@ -296,9 +309,23 @@ class indexDefinition {
     }
 
+    private schemaDefinitionsToDto(): Record<string, string> {
+        if (!this.schemaDefinitions().length) {
```
posible NRE? IDK how knockout handles observableArray

M4xymm replied:
I added an empty array to initialize the variable. I checked this case and it worked.

kalczur commented on src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts:
```diff
@@ -1262,6 +1270,24 @@ class editIndex extends shardViewModelBase {
+
+    formatSchemaDefinition(schemaItem: schemaDefinitionModel) {
+        try {
+            schemaItem.schemaText(generalUtils.prettifyContent(schemaItem.schemaText()));
```
prettifyContent never throws

kalczur commented on src/Raven.Studio/typescript/models/database/index/schemaDefinitionModel.ts:
```diff
@@ -0,0 +1,51 @@
+        return new schemaDefinitionModel();
+    }
+
+    toDto(): { collectionName: string; schemaText: string } {
```
it's not used

M4xymm committed:
RavenDB-25594 refactor (schema definitions): Simplify schema handling and improve JSON formatting

- Removed unused `toDto` method in `schemaDefinitionModel`.
- Renamed `prettifyContent` to `prettifyJson` for clarity.
- Updated index definitions to use default values for `schemaDefinitions`.
- Refined schema formatting logic to reduce redundancy.
